flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -435,6 +435,7 @@ private boolean checkAvailability() {
                 return unavailableSubpartitionsCount == 0;
             } else {
                 requestMemorySegmentFromGlobalWhenAvailable();
+                return !availableMemorySegments.isEmpty() && unavailableSubpartitionsCount == 0;
```

Comment:
Could you maybe extract this check to maybe something like `boolean shouldBeAvailable()`? I think it is duplicated at least once (maybe more) in `checkConsistentAvailability()` method.